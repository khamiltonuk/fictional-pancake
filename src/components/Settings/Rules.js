// @flow
import React, { Fragment } from "react";
import styled from "styled-components";

const Title = styled.h2`
  font-size: 100%;
  font-weight: normal;
`;

const Rules = () => {
  return (
    <Fragment>
      <Title>Rules</Title>
      <p>
        "It's very simple. Scissors cuts paper, paper covers rock, rock crushes
        lizard, lizard poisons Spock, Spock smashes scissors, scissors
        decapitates lizard, lizard eats paper, paper disproves Spock, Spock
        vaporizes rock, and as it always has rock crushes scissors"
      </p>
    </Fragment>
  );
};

export default Rules;
