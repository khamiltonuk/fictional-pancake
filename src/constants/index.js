export const RESET_GAME = "RESET_GAME";
export const SHOW_SETTINGS = "SHOW_SETTINGS";
export const HIDE_SETTINGS = "HIDE_SETTINGS";
export const TOGGLE_AI_MODE = "TOGGLE_AI_MODE";
export const SET_GAME_MODE = "SET_GAME_MODE";
