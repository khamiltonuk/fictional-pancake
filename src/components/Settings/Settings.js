import React from "react";

import Rules from "./Rules";

const Settings = () => {
  return (
    <div id="settings-pane" className="settings" role="dialog">
      <form>
        <fieldset>
          <h2>Select you game mode</h2>
          <div className="inputfield">
            <input
              type="radio"
              name="game-mode"
              value="3"
              id="best-of-3"
              defaultChecked
            />
            <label htmlFor="best-of-3">
              <span />Best of 3
            </label>
          </div>
          <div className="inputfield">
            <input type="radio" name="game-mode" value="5" id="best-of-5" />
            <label htmlFor="best-of-5">
              <span />Best of 5
            </label>
          </div>
          <div className="inputfield">
            <input type="radio" name="game-mode" value="0" id="non-stop" />
            <label htmlFor="non-stop">
              <span />Non stop
            </label>
          </div>
        </fieldset>
        {/* <fieldset>
                            <h2>Select you game type</h2>
                            <div className="inputfield">
                                <input type="radio" name="game-type" value="rps" id="rps"/>
                                <label htmlFor="rps"><span></span>Rock paper scissors</label>
                            </div>
                            <div className="inputfield">
                                <input type="radio" name="game-type" value="rpsls" id="rpsls"  defaultChecked/>
                                <label htmlFor="rpsls"><span></span>Rock paper scissors lizard spock</label>
                            </div>
                        <fieldset> */}
        <Rules />
      </form>
    </div>
  );
};
export default Settings;
