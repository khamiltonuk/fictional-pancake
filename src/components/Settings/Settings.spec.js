import React from "react";
import { shallow, simulate } from "enzyme";
import Settings from "./Settings";
import toJson from "enzyme-to-json";
import { EHOSTUNREACH } from "constants";

describe("Settings", () => {
  test("should match the snapshot", () => {
    const wrapper = shallow(<Settings />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  describe("GameLength", () => {
    test("should call setGameLength function when best-of-3 button is pressed", () => {
      const props = {
        setGameLength: jest.fn()
      };
      const wrapper = shallow(<Settings {...props} />);
      const button = wrapper.find("#best-of-3");
      button.simulate("change", { target: { value: "3" } });
      expect(props.setGameLength).toHaveBeenCalledWith("3");
    });

    test("should have correct GameLength radio input if 3 selected", () => {
      const props = {
        gameLength: "3"
      };
      const wrapper = shallow(<Settings {...props} />);
      const bestOfThreeInput = wrapper.find("#best-of-3");
      const bestOfFiveInput = wrapper.find("#best-of-5");
      const nonStopInput = wrapper.find("#non-stop");
      expect(bestOfThreeInput.props().checked).toEqual(true);
      expect(bestOfFiveInput.props().checked).toEqual(false);
      expect(nonStopInput.props().checked).toEqual(false);
    });

    test("should have correct GameLength radio input if 5 selected", () => {
      const props = {
        gameLength: "5"
      };
      const wrapper = shallow(<Settings {...props} />);
      const bestOfThreeInput = wrapper.find("#best-of-3");
      const bestOfFiveInput = wrapper.find("#best-of-5");
      const nonStopInput = wrapper.find("#non-stop");
      expect(bestOfThreeInput.props().checked).toEqual(false);
      expect(bestOfFiveInput.props().checked).toEqual(true);
      expect(nonStopInput.props().checked).toEqual(false);
    });
  });

  describe("GameMode", () => {
    test("should call setGameMode function when rps button is pressed", () => {
      const props = {
        setGameMode: jest.fn()
      };
      const wrapper = shallow(<Settings {...props} />);
      const button = wrapper.find("#rps");
      button.simulate("change", { target: { value: "rps" } });
      expect(props.setGameMode).toHaveBeenCalledWith("rps");
    });

    test("should have correct GameMode radio input if rpsls selected", () => {
      const props = {
        gameMode: "rpsls"
      };
      const wrapper = shallow(<Settings {...props} />);
      const rpslsInput = wrapper.find("#rpsls");
      const rpsInput = wrapper.find("#rps");
      expect(rpslsInput.props().checked).toEqual(true);
      expect(rpsInput.props().checked).toEqual(false);
    });

    test("should have correct GameMode radio input if rps selected", () => {
      const props = {
        gameMode: "rps"
      };
      const wrapper = shallow(<Settings {...props} />);
      const rpslsInput = wrapper.find("#rpsls");
      const rpsInput = wrapper.find("#rps");
      expect(rpslsInput.props().checked).toEqual(false);
      expect(rpsInput.props().checked).toEqual(true);
    });
  });

  test("onSubmit expect to hide Settings panel", () => {
    const props = {
      hideSettings: jest.fn()
    };
    const wrapper = shallow(<Settings {...props} />);
    wrapper.find("form").simulate("submit", { preventDefault: jest.fn() });
    expect(props.hideSettings).toHaveBeenCalled();
  });
});
