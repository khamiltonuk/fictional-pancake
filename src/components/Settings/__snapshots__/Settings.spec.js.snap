// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings should match the snapshot 1`] = `
<div
  className="settings"
  id="settings-pane"
>
  <form
    method="POST"
    onSubmit={[Function]}
  >
    <styled.fieldset>
      <styled.h2>
        Select you game mode
      </styled.h2>
      <styled.div>
        <styled.input
          checked={false}
          id="best-of-3"
          name="game-mode"
          onChange={[Function]}
          type="radio"
          value="3"
        />
        <styled.label
          htmlFor="best-of-3"
        >
          <styled.span />
          Best of 3
        </styled.label>
      </styled.div>
      <styled.div>
        <styled.input
          checked={false}
          id="best-of-5"
          name="game-mode"
          onChange={[Function]}
          type="radio"
          value="5"
        />
        <styled.label
          htmlFor="best-of-5"
        >
          <styled.span />
          Best of 5
        </styled.label>
      </styled.div>
      <styled.div>
        <styled.input
          checked={false}
          id="non-stop"
          name="game-mode"
          onChange={[Function]}
          type="radio"
          value="0"
        />
        <styled.label
          htmlFor="non-stop"
        >
          <styled.span />
          Non stop
        </styled.label>
      </styled.div>
    </styled.fieldset>
    <styled.fieldset>
      <styled.h2>
        Select you game type
      </styled.h2>
      <styled.div>
        <styled.input
          checked={false}
          id="rps"
          name="game-type"
          onChange={[Function]}
          type="radio"
          value="rps"
        />
        <styled.label
          htmlFor="rps"
        >
          <styled.span />
          Rock paper scissors
        </styled.label>
      </styled.div>
      <styled.div>
        <styled.input
          checked={false}
          id="rpsls"
          name="game-type"
          onChange={[Function]}
          type="radio"
          value="rpsls"
        />
        <styled.label
          htmlFor="rpsls"
        >
          <styled.span />
          Rock paper scissors lizard spock
        </styled.label>
      </styled.div>
    </styled.fieldset>
    <styled.button
      type="submit"
    >
      Done
    </styled.button>
    <Rules />
  </form>
</div>
`;
